.github/workflows/client-build.yaml

```diff
@@ -10,19 +10,13 @@ on:
     paths:
       - '.github/workflows/**'
       - 'api/**'
-      - 'client/flutter/pubspec.yaml'
-      - 'client/flutter/pubspec.lock'
-      - 'client/flutter/ios/**'
-      - 'client/flutter/android/**'
+      - 'client/flutter/**'
```

Comment:
Do we really want to run the build step even if just some Dart code has changed? cc @SamMousa

Reply:
We do if we want to build the whole app every time; we don't need to if we just care if the apps build. (Since dart code should be tested outside the app build via `flutter test`)

Reply:
Without this change, you fail to test that no changes were required to the ios/android specific code or gradle/cocoapods.  Every client change should result in a working iOS/Android build.